lib/liquid/expression.rb

```diff
@@ -25,7 +25,8 @@ def self.parse(markup)
       when INTEGERS_REGEX
         Regexp.last_match(1).to_i
       when RANGES_REGEX
-        RangeLookup.parse(Regexp.last_match(1), Regexp.last_match(2))
+        parts = markup.strip.gsub(/\A\(|\)\Z/, '').split(/\.{2,}/)
```

Comment:
`/\.{2,}/` because the old regex supported `(1............2)` and there's actually a test to make sure this "quirk" is supported.